src/ParticleDA.jl

```diff
@@ -616,91 +266,91 @@ function tdac(params::Parameters, rng::AbstractVector{<:Random.AbstractRNG})
             end
 
         end
 
     end
 
-    if params.enable_timers
+    if filter_params.enable_timers
 
-        if my_rank == params.master_rank
+        if my_rank == filter_params.master_rank
             print_timer(timer)
         end
 
-        if params.verbose
+        if filter_params.verbose
             # Gather string representations of timers from all ranks and write them on master
             str_timer = string(timer)
 
             # Assume the length of the timer string on master is the longest (because master does more stuff)
-            if my_rank == params.master_rank
+            if my_rank == filter_params.master_rank
                 length_timer = length(string(timer))
             else
                 length_timer = nothing
             end
 
-            length_timer = MPI.bcast(length_timer, params.master_rank, MPI.COMM_WORLD)
+            length_timer = MPI.bcast(length_timer, filter_params.master_rank, MPI.COMM_WORLD)
 
             chr_timer = Vector{Char}(rpad(str_timer,length_timer))
 
-            timer_chars = MPI.Gather(chr_timer, params.master_rank, MPI.COMM_WORLD)
+            timer_chars = MPI.Gather(chr_timer, filter_params.master_rank, MPI.COMM_WORLD)
 
-            if my_rank == params.master_rank
-                write_timers(length_timer, my_size, timer_chars, params)
+            if my_rank == filter_params.master_rank
+                # write_timers(length_timer, my_size, timer_chars, filter_params)
             end
         end
     end
 
     unpack_statistics!(avg_arr, var_arr, statistics)
 
-    return states.truth, avg_arr, var_arr
+    return get_truth(model_data), avg_arr, var_arr
 end
 
 # Initialise params struct with user-defined dict of values.
-function get_params(user_input_dict::Dict)
+function get_params(T, user_input_dict::Dict)
 
     user_input = (; (Symbol(k) => v for (k,v) in user_input_dict)...)
-    params = Parameters(;user_input...)
+    params = T(;user_input...)
 
 end
 
+get_params(user_input_dict::Dict) = get_params(FilterParameters, user_input_dict)
+
 # Initialise params struct with default values
-get_params() = Parameters()
+get_params() = FilterParameters()
 
-function get_params(path_to_input_file::String)
+function read_input_file(path_to_input_file::String)
 
     # Read input provided in a yaml file. Overwrite default input parameters with the values provided.
     if isfile(path_to_input_file)
         user_input_dict = YAML.load_file(path_to_input_file)
-        params = get_params(user_input_dict)
-        if params.verbose
-            println("Read input parameters from ",path_to_input_file)
-        end
     else
         @warn "Input file " * path_to_input_file * " not found, using default parameters"
-        params = get_params()
+        user_input_dict = Dict()
     end
-    return params
+    return user_input_dict
 
 end
 
-function tdac(path_to_input_file::String, rng::AbstractRNG)
```

Comment:
The diff is screwing things up a bit, what I did here was to remove this method and keep only the two other methods below